### DISCUSSION TITLE
[FR] Protected branches: Require _involvement_ from Code Owners

### DISCUSSION THREAD

**Discussion by ARR4N** [community/community#37120] [Pull Requests]
**Background**: There's currently a setting to _Require review from Code Owners_ on protected branches. For some teams this can prove unnecessarily cumbersome when the PR author is also an owner and the team policy would otherwise be that they only require a second set of eyes.

**Feature request**: A mutually exclusive, less strict setting _Require involvement from Code Owners_ that would be met by an author or reviewer being an owner as per the existing `CODEOWNERS`. This is not appropriate for all scenarios, so the original setting would remain.

**Example scenario**: Alice and Bob are the sole owners of code `/foo/**`; Charlie is another SWE. Alice (i.e. an owner) sends a PR that under the current setting MUST be reviewed by Bob even if Charlie has approved it. Under the proposed setting, Alice / Bob wouldn't require additional approval of changes to `/foo/.*` unless _Require approvals_ is set >0, but Charlie would still require a review from at least Alice / Bob.

**Prior art**: this is how Google's monorepo enforces ownership. That rationale is that code _ownership_ is to ensure that broader contextual needs are taken into account when modifying code whereas code _review_ is for the change at hand.

**Comment by ossanna16** [comment]
Hi there @aschlosberg and welcome to our community! Thank you for your suggestion 🙂
